react-backend/functions/pdfgenerator.js

```diff
@@ -22,46 +86,32 @@ module.exports = QRcodeBase64Url => {
 
     content: [
       {
-        columns: [
-          {
-            image: `${__dirname}/../public/qrcodelogo.png`,
-            width: 38,
-            height: 129.5,
-            margin: [0, 10, 0, 0],
-          },
-          {
-            image: QRcodeBase64Url,
-            margin: [25, 5, 0, 0],
-            fit: [125, 125],
-          },
-        ],
+        columns,
       },
       {
         text: 'Use this QR Code to sign in',
         fontSize: 14,
-        alignment: 'right',
-        margin: [0, -10, 10, 0],
+        alignment: textAlignment,
+        margin: textMargin,
         color: '#0a092a',
       },
     ],
   };
 
-  return new Promise((resolve, reject) => {
-    const printer = new PdfPrinter(fontDescriptors);
+  const printer = new PdfPrinter(fontDescriptors);
 
-    const doc = printer.createPdfKitDocument(docDefinition);
+  const doc = printer.createPdfKitDocument(docDefinition);
 
-    const chunks = [];
-    let result;
+  const chunks = [];
+  let result;
 
-    doc.on('data', chunk => {
-      chunks.push(chunk);
-    });
-    doc.on('end', () => {
-      result = Buffer.concat(chunks);
-      resolve(result.toString('base64'));
-      reject("Couldn't generate PDF");
-    });
-    doc.end();
+  doc.on('data', chunk => {
+    chunks.push(chunk);
+  });
+  doc.on('end', () => {
+    result = Buffer.concat(chunks);
```

Comment:
Result can just be a `const` defined down here.

Reply:
I think the whole thing can be

```js
  const printer = new PdfPrinter(fontDescriptors);
  const doc = printer.createPdfKitDocument(docDefinition);

  const chunks = [];

  doc.on('data', chunk => {
    chunks.push(chunk);
  });

  doc.on('end', () => {
    resolve(Buffer.concat(chunks).toString('base64'));
  });
  doc.end();
```

Reply:
done :)